athena-federation-sdk/src/test/java/com/amazonaws/athena/connector/lambda/handlers/GlueMetadataHandlerTest.java

```diff
@@ -192,38 +195,96 @@ public void doListTables()
                     }
                     else {
-                        //only return real info on 1st call
-                        when(mockResult.getTableList()).thenReturn(new ArrayList<>());
-                        when(mockResult.getNextToken()).thenReturn(null);
+                        // Simulate paginated list of tables returned from Glue.
+                        List<Table> paginatedTables = tables.stream()
+                                .sorted(Comparator.comparing(Table::getName))
+                                .filter(table -> nextToken == null || table.getName().compareTo(nextToken) >= 0)
+                                .limit(pageSize + 1)
+                                .collect(Collectors.toList());
+                        if (paginatedTables.size() > pageSize) {
+                            when(mockResult.getNextToken()).thenReturn(paginatedTables.get(pageSize).getName());
+                            when(mockResult.getTableList()).thenReturn(paginatedTables.subList(0, pageSize));
+                        }
+                        else {
+                            when(mockResult.getNextToken()).thenReturn(null);
+                            when(mockResult.getTableList()).thenReturn(paginatedTables);
+                        }
                     }
                     return mockResult;
                 });
 
-        ListTablesRequest req = new ListTablesRequest(IdentityUtil.fakeIdentity(), queryId, catalog, schema);
-        ListTablesResponse res = handler.doListTables(allocator, req);
-        logger.info("doListTables - {}", res.getTables());
-
-        Set<String> tableNames = tables.stream().map(next -> next.getName()).collect(Collectors.toSet());
-        for (TableName next : res.getTables()) {
-            assertEquals(schema, next.getSchemaName());
-            assertTrue(tableNames.contains(next.getTableName()));
-        }
-        assertEquals(tableNames.size(), res.getTables().size());
+        logger.info("doListTables - Unlimited page size");
+        ListTablesRequest req = new ListTablesRequest(IdentityUtil.fakeIdentity(),
+                queryId, catalog, schema, null, UNLIMITED_PAGE_SIZE_VALUE);
+        logger.info("doListTables - {}", req);
+        ListTablesResponse actualResponse = handler.doListTables(allocator, req);
+        logger.info("doListTables - {}", actualResponse);
+        assertEquals("Lists do not match.", fullListResponse, actualResponse);
+
+        logger.info("doListTables - Large page-size request");
+        req = new ListTablesRequest(IdentityUtil.fakeIdentity(),
+                queryId, catalog, schema, null, GET_TABLES_REQUEST_MAX_RESULTS + 50);
+        logger.info("doListTables - {}", req);
+        actualResponse = handler.doListTables(allocator, req);
+        logger.info("doListTables - {}", actualResponse);
+        assertEquals("Lists do not match.", fullListResponse, actualResponse);
```

Comment:
should these be different tests? this one test is getting rather large.

Reply:
moved some of the mock dependencies and reduced the size substantially.